### PR TITLE
Feature/cs4360 361 ng account service

### PR DESCRIPTION
**Pull request by GitNickProgramming**

### CODE REVIEW

File: pb-crm-ui/src/app/invoicing/services/account.service.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import {
+  AccountModel,
+  CreateAccountRequest,
+  CreateAccountResponse, UpdateAccountRequest, UpdateAccountResponse
+} from '@hiddentemple/api-interfaces';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ContactService {
```
Comment (Newms226) [inline pb-crm-ui/src/app/invoicing/services/account.service.ts:15]: All the routes in this class are pointing to contact. Also this class is not named correctly.

File: pb-crm-ui/src/app/invoicing/services/account.service.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import {Injectable} from '@angular/core';
+import {Observable} from 'rxjs';
+import {ApiService} from '../../api/api.service';
+import {GetAllAccountResponse} from '@hiddentemple/api-interfaces';
+import {GetAccountResponse} from '@hiddentemple/api-interfaces';
+import {
+  AccountModel,
+  CreateAccountRequest,
+  CreateAccountResponse, UpdateAccountRequest, UpdateAccountResponse
+} from '@hiddentemple/api-interfaces';
```
Comment (Newms226) [inline pb-crm-ui/src/app/invoicing/services/account.service.ts:10]: You should run an organize imports to clean this up

File: pb-crm-ui/src/app/invoicing/services/account.service.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    if (Array.isArray(value) && value.length === 0) { return acc; }
+    else if (typeof value === 'string' && value.length === 0) { return acc; }
+    else if (value === {}) {return acc; }
+    else { return {...acc, [key]: value}; }
+  }
+
+  reduceToDefined<T>(model: T): Partial<T> {
+    const acc: Partial<T> = {};
+    return Object.entries(model).reduce(this.emptyReducer, acc);
+  }
```
Comment (Newms226) [inline pb-crm-ui/src/app/invoicing/services/account.service.ts:36]: Remove all this functionality please

**Review by Newms226 — CHANGES_REQUESTED** [review]
Please address my comments and send a message in discord once fixed.

Looks great :)

**Comment by GitNickProgramming** [comment]
Resubmitting

**Pull request closed without merging**